src/api.js

```diff
@@ -0,0 +1,187 @@
+* @param{string} id the app id
+* @returns{string} the table name in SQL table format
+*/
+const getTable = function (map, id) {
+  const regex = /(^[a-z]{2,3})([\.][a-z0-9]*)+$/gmi;
+  if (!id.match(regex)) {
+    const msg = `cannot parse app id: '${id}'. Check formatting and try again`;
+    throw new MalformedArgumentError(msg);
+  }
+  const obj = map[id];
+  if (!obj ) {
+    const msg = `could not find a table for app id: '${id}'`;
+    const err = new Error(msg);
+    throw err;
+  }
+  return `${obj.project}.${obj.dataset}.${obj.table}`;
+}
+
+function sendRows(res, rows, nextCursor, key) {
+  if (!rows || rows.length == 0 || !nextCursor) {
+    cacheManager.removeCache(key);
+  }
+  const resObj = formatRowsToJson(rows);
+  return res.status(200).json({data: resObj, nextCursor});
+}
+
+function formatRowsToJson (rows) {
+  let resObj = [];
+  if (!rows || rows.length === 0) return resObj;
+  rows.forEach((row) => {
+    resObj.push({
+      attribution_url: row.referral_source,
+      app_id: row.app_package_name,
+      ordered_id: row.event_timestamp,
+      user: {
+        id: row.user_pseudo_id,
+        metadata: {
+          continent: row.continent,
+          country: row.country,
+          region: row.region,
+          city: row.city,
+        },
+        ad_attribution: {
+          source: getSource(row.referral_source),
+          data: {
+            advertising_id: row.advertising_id,
+          },
+        },
+        event: {
+          name: row.event_name,
+          date: row.event_date,
+          timestamp: row.event_timestamp,
+          label: row.label,
+          action: row.action,
+          value: row.value,
+          value_type: row.type,
+        }
+      }
+
+    });
+  });
+  return resObj;
+}
+
+function getSource(referralString) {
+  if (!referralString) return "no-source";
+  const regex = /^([a-z]{1,6})(_[a-z]{1,3})?/gmi;
+  const group = referralString.match(regex);
+  switch (group.toString().toLowerCase()) {
+    case 'fb_web':
+      return 'fb_web';
+    case 'fb_app':
+      return 'fb_app';
+    case 'google_web':
+      return 'google_web';
+    case 'google_app':
+      return 'google_app';
+    case 'direct':
+      return 'direct';
+    default:
+      return 'no-source';
+  }
+}
+
+/**
+* handler for GET requests to /fetch_latest
+* returns all records for app_id logged after the given cursor
+* @param{obj} req the request
+* @param{obj} res the response
+*/
+async function fetchLatestHandler (req, res, next) {
+  const searchParams = req.query;
+
+  if (!searchParams.app_id) {
+    return res.status(400).send({msg: config.errNoId});
+  } else if (!searchParams.from) {
+    return res.status(400).send({msg: config.errNoCursor});
+  } else if (!searchParams.from.match(/[A-Z0-9]+/gmi)) {
+    return res.status(400).send({msg: config.errBadCursor});
+  }
+  const sql = fs.readFileSync(config.fetchLatestQuery.string).toString();
```

Comment:
As a small performance rule, we probably don't want to hit the disk every time we get a request. Why not just read this once (on start up) and then provide it? In other words, create a module that reads this and exports the in-memory string? 

Remember, node modules are functionally singletons, so whenever you have some process of "run once, expose forever", you can just use a module!